src/OrchardCore/OrchardCore.Users.Abstractions/Services/IClaimsProvider.cs

```diff
@@ -0,0 +1,10 @@
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Users.Services
+{
+    public interface IClaimsProvider
+    {
+        Task<ClaimsIdentity> GenerateAsync(IUser user, ClaimsIdentity claims);
```

Comment:
```suggestion
        Task GenerateAsync(IUser user, ClaimsIdentity claims);
```

I don't think it makes sense to return them as well.